media_service/tests/test_iiif.py

```diff
@@ -28,20 +28,40 @@ def setUp(self):
         self.factory = RequestFactory()
 
     def create_manifest(self, manifest_id, **kwargs):
-        request = self.factory.get('/')
-        return request, IIIFManifest(manifest_id, **kwargs)
+        return IIIFManifest(manifest_id, **kwargs)
 
     def get_images_list(self):
         images = [
             {'id': 1, 'is_link': False, 'url': 'http://localhost:8000/loris/foo.jpg', 'label': 'foo.jpg', 'description': '', 'metadata': []},
             {'id': 2, 'is_link': False, 'url': 'http://localhost:8000/loris/bar.jpg', 'label': 'bar.jpg', 'description': '', 'metadata': []},
             {'id': 3, 'is_link': True, 'url': 'http://my.link/image.jpg', 'label': 'image.jpg', 'description': '', 'metadata': []},
+            {
+                'id': 4,
+                'is_link': False,
+                'url': 'http://localhost:8000/loris/foo_bar_foo.png',
+                'label': 'foo_bar_foo.png',
+                'description': 'Foo Bar with Foo Again',
+                'metadata': [
+                    {"label": "Date", "value": "Once upon a time..."},
+                    {"label": "Location", "value": "In a galaxy far far away..."},
+                ]
+            },
         ]
         return images
+    
+    def duplicate_image(self, images, image_id, num_duplicates=1):
+        '''
+        Given a list of images, find an image by ID and duplicate it.
+        '''
+        image = next(iter([i for i in images if i['id'] == image_id]))
```

Comment:
Interesting. I would've written this as, e.g.
```
image = [i for i in images if i['id'] == image_id][0]
```
which would raise an `IndexError` instead of a `StopIteration`. I wasn't familiar with this idiom -- any particular advantages, outside of the case where you have many, many thousands of `i in images` where `i['id'] == image_id`? (That is probably enough of an advantage to switch to using this idiom, just wondering.)